packages/core/src/request-builder/header-builder/header-builder.ts

```diff
@@ -21,89 +21,80 @@ import { addCsrfTokenAndCookies } from './csrf-token-header';
  * @param request - OData request configuration to create headers for
  * @returns Key-value pairs where the key is the name of a header property and the value is the respective value
  */
-export function buildHeaders<RequestT extends ODataRequestConfig>(
+export async function buildHeaders<RequestT extends ODataRequestConfig>(
   request: ODataRequest<RequestT>
 ): Promise<MapType<string>> {
   if (!request.destination) {
     throw Error('The request destination is undefined.');
   }
-  return buildHeadersAuthFirstCustomLast(request)(
-    addEtagHeader(request.config),
-    headers => addCsrfTokenAndCookies(request, headers),
-    addAcceptHeader,
-    headers => addContentTypeHeader(request.config.contentType, headers),
-    addSapClientHeader(request.destination.sapClient || undefined),
-    addProxyHeaders(request.destination.proxyConfiguration),
-    addLocationIdHeader(request.destination.cloudConnectorLocationId)
+
+  const defaultHeaders = replaceDuplicateKeys(
+    filterNullishValues({
+      Accept: 'application/json',
+      'Content-Type': request.config.contentType,
+      'if-match': getETagHeaderValue(request.config)
+    }),
+    request.config.customHeaders
```

Comment:
Is it possible to use the spread `...` to replace the  keys from the first object with the ones from the customHeaders object?

Reply:
No because this does not take into consideration upper and lower case, e. g. if customHeaders contains a 'accept' header.